include/XLink/XLinkPrivateDefines.h

```diff
@@ -83,35 +83,42 @@ typedef enum
 {
     /*USB-X_LINK_PCIE related events*/
     XLINK_WRITE_REQ,
+    XLINK_WRITE_FD_REQ, // only for the shared mem protocol
     XLINK_READ_REQ,
     XLINK_READ_REL_REQ,
+    XLINK_READ_REL_SPEC_REQ,
     XLINK_CREATE_STREAM_REQ,
     XLINK_CLOSE_STREAM_REQ,
     XLINK_PING_REQ,
     XLINK_RESET_REQ,
+
     XLINK_REQUEST_LAST,
     //note that is important to separate request and response
     XLINK_WRITE_RESP,
+    XLINK_WRITE_FD_RESP, // only for the shared mem protocol
     XLINK_READ_RESP,
     XLINK_READ_REL_RESP,
+    XLINK_READ_REL_SPEC_RESP,
     XLINK_CREATE_STREAM_RESP,
     XLINK_CLOSE_STREAM_RESP,
     XLINK_PING_RESP,
     XLINK_RESET_RESP,
+
     XLINK_RESP_LAST,
 
     /*X_LINK_IPC related events*/
     IPC_WRITE_REQ,
+    IPC_WRITE_FD_REQ,
     IPC_READ_REQ,
     IPC_CREATE_STREAM_REQ,
     IPC_CLOSE_STREAM_REQ,
+
     //
     IPC_WRITE_RESP,
+    IPC_WRITE_FD_RESP,
     IPC_READ_RESP,
     IPC_CREATE_STREAM_RESP,
     IPC_CLOSE_STREAM_RESP,
-    XLINK_READ_REL_SPEC_REQ,
-    XLINK_READ_REL_SPEC_RESP,
```

Comment:
Why removed?

Also, all new events MUST GO to the end of enum, otherwise backward compatibility is broken

Reply:
I moved them before XLINK_LAST_REQ and XLINK_LAST_RESP, otherwise they will cause this assert to fail:
```
int handleIncomingEvent(xLinkEvent_t* event, XLinkTimespec treceive) {
    //this function will be dependent whether this is a client or a Remote
    //specific actions to this peer
    mvLog(MVLOG_DEBUG, "%s, size %u, streamId %u.\n", TypeToStr(event->header.type), event->header.size, event->header.streamId);

    ASSERT_XLINK(event->header.type >= XLINK_WRITE_REQ
               && event->header.type != XLINK_REQUEST_LAST
               && event->header.type < XLINK_RESP_LAST); 
```